keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationSupport.kt

```diff
@@ -100,7 +101,8 @@ class AuthorizationSupport(
     withAuthentication(target, identifier) { auth ->
       val application = when (target) {
         RESOURCE -> repository.getResource(identifier).application
-        APPLICATION, DELIVERY_CONFIG -> identifier
+        APPLICATION -> identifier
+        DELIVERY_CONFIG -> getDeliveryConfigApplication(identifier)
```

Comment:
This looks OK to me, but I don't understand why you would fall back to the identifier if you fail to find the delivery config by name... There's no guarantee that the identifier will match the application.

I'd suggest just reverting back to the original behavior:
```kotlin
DELIVERY_CONFIG -> repository.getDeliveryConfig(identifier).application
```
Or am I missing something?

Reply:
it can be either the name of the application OR the name of the delivery config.. so I'm not quite sure how `repository.getDeliveryConfig(identifier).application` would work if you pass the application name and not the delivery config name

Reply:
Hmm... But you specify which one to use in the annotation... There shouldn’t be a need to do this unless there’s a controller parameter that can be _either_ the delivery config name _or_ the application name, but I don’t recall seeing anything like that.

Reply:
Hmm... But you specify which one to use in the annotation... There shouldn’t be a need to do this unless there’s a controller parameter that can be _either_ the delivery config name _or_ the application name, but I don’t recall seeing anything like that.